### PR TITLE
Utilities plugin to plot CDC pulse height vs pulse time

### PR DESCRIPTION
**Pull request by nsjarvis**

### CODE REVIEW

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/make_nsj_cdc_amp_t.log](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/make_nsj_cdc_amp_t.log)
 Build report: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/report_nsj_cdc_amp_t.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/report_nsj_cdc_amp_t.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t

File: src/plugins/Utilities/SConscript [hunk 1]
```diff
@@ -14,5 +14,5 @@ SConscript(dirs=subdirs, exports='env osname', duplicate=0)
 # Optional targets
 optdirs = ['danahddm', 'dumpcandidates', 'dumpthrowns', 'l3bdt']
 optdirs.extend(['merge_rawevents', 'syncskim', 'DAQ', 'TTab', 'rawevent'])
-optdirs.extend(['cdc_emu', 'fmwpc_scan', 'epem_ml_skim'])
+optdirs.extend(['cdc_amp_t','cdc_echo','cdc_emu', 'fmwpc_scan', 'epem_ml_skim'])
```
Comment (aaust) [inline src/plugins/Utilities/SConscript:17]: cdc_echo cannot be a default target and an optional target at the same time

File: src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+}
+
+//------------------
+// init
+//------------------
+jerror_t JEventProcessor_cdc_amp_t::init(void)
+{
+	// This is called once at program startup. 
+
+    amp_t = new TH2I("amp_t","Digihit pulse height - pedestal vs sample number;sample number;pulse height-pedestal", 200,0,200,205,0,4100);
```
Comment (aaust) [inline src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc:64]: Both cdc_echo and cdc_amp_t have histograms with this name, so there is a conflict when running both plugins at the same time. It is a good practice to create a folder with the plugin name in the histogram root file.

File: src/plugins/Utilities/SConscript (hunk 1 above)
Comment (nsjarvis) [reply src/plugins/Utilities/SConscript:17]: Sorry.

File: src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc (hunk 2 above)
Comment (nsjarvis) [reply src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc:64]: Okay.  They are the same histogram, both plugins have a very specific use case.  I guess I could delete the histogram from inside cdc_echo.

File: src/plugins/Utilities/SConscript (hunk 1 above)
Comment (aaust) [reply src/plugins/Utilities/SConscript:17]: I removed cdc_echo from the default targets, as it serves only a 'specific use case'

File: src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc (hunk 2 above)
Comment (aaust) [reply src/plugins/Utilities/cdc_amp_t/JEventProcessor_cdc_amp_t.cc:64]: Ok, the policy does not strictly apply to optional plugins.

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/make_nsj_cdc_amp_t.log](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/make_nsj_cdc_amp_t.log)
 Build report: [/work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t/report_nsj_cdc_amp_t.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^nsj_cdc_amp_t/report_nsj_cdc_amp_t.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^nsj_cdc_amp_t

**Review by aaust — APPROVED** [review]
The plugin is an optional target, so I just made sure it compiled.

**Pull request merged**